scene/2d/animated_sprite_2d.cpp

```diff
@@ -172,17 +172,18 @@ void AnimatedSprite2D::_notification(int p_what) {
 				return;
 			}
 
-			double speed = frames->get_animation_speed(animation) * Math::abs(speed_scale);
-			if (speed == 0) {
-				return; // Do nothing.
-			}
-			int last_frame = frames->get_frame_count(animation) - 1;
-
 			double remaining = get_process_delta_time();
+			int i = 0;
 			while (remaining) {
-				if (timeout <= 0) {
-					timeout = _get_frame_duration();
+				if (speed_scale == 0) {
+					return; // Do nothing.
+				}
+
+				// Frame count may be changed by animation_finished or frame_changed signals.
+				int fc = frames->get_frame_count(animation);
 
+				if (timeout <= 0) {
+					int last_frame = fc - 1;
```

Comment:
If it don't need the fc check, you can change this.
```suggestion
				if (timeout <= 0) {
					// Frame count may be changed by animation_finished or 
					int last_frame = frames->get_frame_count(animation) - 1;
```